flink-tests/src/test/java/org/apache/flink/test/state/operator/restore/unkeyed/ChainLengthIncreaseTest.java

```diff
@@ -64,4 +64,8 @@ public void createRestoredJob(StreamExecutionEnvironment env) {
         SingleOutputStreamOperator<Integer> third =
                 createThirdStatefulMap(ExecutionMode.RESTORE, stateless2);
     }
+
+    /** Will use the state generate by {@link ChainBreakTest}. */
+    @Override
+    public void generateSnapshots(FlinkVersion targetVersion) throws Exception {}
```

Comment:
Having all these empty classes feels odd. I didn't go through the code in detail. But it looks like we need to have it implemented by `ChainBreakTest` and been disabled by all the others so that the `AbstractOperatorRestoreTestBase.generateSnapshots` method is only called once? Could we extract the data generation out of `AbstractOperatorRestoreTestBase`, instead? :thinking: